File: backend/tiramisu-lite/DTO/PlanerItemDto.cs
namespace tiramisu_lite.DTO;

public record PlanerItemDto
{
    public Guid Id { get; init; }
    public string Title { get; init; }
    public DateTime EatTime { get; init; }
    public IEnumerable<MealDto> Meals { get; init; }
    public decimal KcalSummary { get; init; }
    public bool Notify { get; init; }
}

File: mobile/lib/model/dto/planer_item_dto.dart
import 'package:mobile/model/dto/meal_dto.dart';

class PlanerItemDto {
  final PlanerItemProps props;
  final List<MealDto> meals;

  PlanerItemDto({required this.props, required this.meals});

  Map<String, dynamic> toJson() {
    return {
      'props': props.toJson(),
      'meals': meals.map((meal) => meal.toJson()).toList(),
    };
  }
}


class PlanerItemProps {
  final String title;
  final DateTime eatDate;
  final bool notify;

  PlanerItemProps({
    required this.title,
    required this.eatDate,
    required this.notify,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'eatDate': eatDate.toIso8601String(),
      'notify': notify,
    };
  }
}

